src/odemis/driver/simsem.py

```diff
@@ -464,12 +464,12 @@ def _simulate_image(self):
             # If the shift caused the image to go out of bounds, limit it
             if ltrb[0] < 0:
                 ltrb[0] = 0
-            elif ltrb[2] > shape[1] - 1:
-                ltrb[0] -= ltrb[2] - (shape[1] - 1)
+            elif ltrb[2] > shape[1]:
+                ltrb[0] -= ltrb[2] - shape[1]
             if ltrb[1] < 0:
                 ltrb[1] = 0
-            elif ltrb[3] >= shape[0] - 1:
-                ltrb[1] -= ltrb[3] - (shape[0] - 1)
+            elif ltrb[3] >= shape[0]:
+                ltrb[1] -= ltrb[3] - shape[0]
```

Comment:
Just a tiny modification without it left and top are equal -1. Tested on all simulators using simsem.